### ISSUE TITLE
Redshift Dialect: Parse error when using GRANT with multiple GROUP targets

### ISSUE THREAD

**Issue by jnjimmy1** [sqlfluff/sqlfluff#7361]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When using the Redshift dialect and linting a GRANT command with multiple GROUP targets, there is a parsing error. Meanwhile, the syntax is valid and able to be run on my Redshift instance.

### Expected Behaviour

The parsing should be able to handle multiple GROUPs being specified. The AWS Redshift documentation specifies that multiple targets can be used.
https://docs.aws.amazon.com/redshift/latest/dg/r_GRANT.html

### Observed Behaviour

With GRANT commands, sqlfluff is able to handle when a single group is specified. However, fails when multiple groups are specified within the same command.

For example:
```sql
-- works
GRANT SELECT, DELETE ON salesshare TO GROUP developer;

-- parse error
GRANT SELECT, DELETE ON salesshare TO GROUP developer, GROUP analyst;
```

### How to reproduce

Create a file named **group_example.sql**
```sql
GRANT SELECT, DELETE ON salesshare TO GROUP developer, GROUP analyst;
```

Run the following lint command:
```
> sqlfluff lint ./group_example.sql --dialect redshift
```

This results in:
```
== [group_example.sql] FAIL                                                                                                     
L:   1 | P:  54 |  PRS | Line 1, Position 54: Found unparsable section: ', GROUP
                       | analyst;'
WARNING: Parsing errors found and dialect is set to 'redshift'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### Dialect

Redshift

### Version

Python 3.11.4
sqlfluff, version 4.0.0a3

### Configuration

default

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
